### PR TITLE
Add enum values generation to documentation

### PR DESCRIPTION
**Pull request by slevenick**
Generate enum possible values and default in Terraform. Remove old handwritten lists of values unless they add value.

Fixes: https://github.com/terraform-providers/terraform-provider-google/issues/6092

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: templates/terraform/property_documentation.erb [hunk 1]
```diff
@@ -14,6 +14,18 @@
 <%   end -%>
 <% end -%>
 <%= indent(property.description.strip.gsub("\n\n", "\n"), 2) -%>
+<% if property.is_a?(Api::Type::Enum) && property.validation.nil? && !property.output -%>
```
Comment (danawillow) [inline templates/terraform/property_documentation.erb:17]: Can we also get this information into the description that appears in the resource schema? I'd imagine that tools that use that information would probably like it to be there too.

Comment (danawillow) [inline templates/terraform/property_documentation.erb:17]: why `property.validation.nil?`?

Comment (slevenick) [reply templates/terraform/property_documentation.erb:17]: Added to the schema in a different format with `"` and `["POSSIBLE_VALUE"]`

File: templates/terraform/property_documentation.erb [hunk 2]
```diff
@@ -14,6 +14,15 @@
 <%   end -%>
 <% end -%>
 <%= indent(property.description.strip.gsub("\n\n", "\n"), 2) -%>
+<% if property.is_a?(Api::Type::Enum) && !property.output -%>
+<% unless property.default_value.nil? || property.default_value == "" -%>
+  Default value: `<%= property.default_value %>`
```
Comment (danawillow) [inline templates/terraform/property_documentation.erb:19]:
```suggestion
  Default value: `<%= property.default_value %>`.
```

File: provider/inspec.rb [hunk 3]
```diff
@@ -306,6 +306,10 @@ def markdown_format(property, indent = 1)
         description_arr += property.item_type.properties.map\
           { |prop| markdown_format(prop, indent + 1) }
         description = description_arr.join("\n\n")
+      elsif property.is_a?(Api::Type::Enum)
+        description_arr = [description, "#{'  ' * indent}Possible values:"]
+        description_arr += property.values.map { |v| "#{'  ' * indent}* #{v}" }
```
Comment (danawillow) [inline provider/inspec.rb:311]: I think you need additional indentation here: https://github.com/modular-magician/inspec-gcp/compare/auto-pr-3404-old..auto-pr-3404?short_path=b6ed1f3#diff-b6ed1f35a7be2989c6939b973a4a6245

File: templates/terraform/property_documentation.erb [hunk 4]
```diff
@@ -14,6 +14,15 @@
 <%   end -%>
 <% end -%>
 <%= indent(property.description.strip.gsub("\n\n", "\n"), 2) -%>
+<% if property.is_a?(Api::Type::Enum) && !property.output -%>
+<% unless property.default_value.nil? || property.default_value == "" -%>
```
Comment (danawillow) [inline templates/terraform/property_documentation.erb:18]: This ends up being a bit weird looking for some attributes where we want to describe the enum meanings (https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-3404-old..auto-pr-3404?short_path=4ab0195#diff-4ab0195b34117c4f006f50cbfb766ae1), but I don't have a great solution there so I'm fine leaving it  as-is. Maybe look out for those and move those descriptions in line with the rest of the field description?

**Pull request merged**